components/sup/src/manager/service/mod.rs

```diff
@@ -370,24 +417,12 @@ impl Service {
     /// Run reconfigure hook if present. Return false if it is not present, to trigger default
     /// restart behavior.
     pub fn reconfigure(&mut self) {
-        match self.config.write() {
-            Ok(true) => {
-                self.needs_restart = true;
-                if let Some(err) = self.run_hook(HookType::Reconfigure).err() {
-                    outputln!(preamble self.service_group,
-                              "Reconfiguration hook failed: {}",
-                              err);
-                }
-                if let Some(err) = self.copy_run().err() {
-                    outputln!(preamble self.service_group, "Failed to copy run hook: {}", err);
-                }
-            }
-            Ok(false) => (),
-            Err(e) => {
-                outputln!(preamble self.service_group,
-                          "Failed to write service configuration: {}",
-                          e);
-            }
+        if let Some(err) = self.run_hook(HookType::Reconfigure).err() {
+            outputln!(preamble self.service_group,
+                      "Reconfiguration hook failed: {}",
+                      err);
+        } else {
+            self.needs_reconfiguration = false;
```

Comment:
Only clearing this bit on run hook success means that the service can get in an infinite restart loop if a bad reconfigure hook is provided. It might be best to just warn and clear this no matter what to try and just keep running if possible. What do you think?